### PR TITLE
Airbyte 8278  all in one static code checker to be run locally as well as during ci pipelines

### LINKED ISSUE #8278: All-in-one static code checker to be run locally as well as during CI pipelines
Subtask for https://github.com/airbytehq/airbyte/issues/8137
At first, we want to run a set of static code checkers as a part of our code quality assurance during CI pipelines.
On the other hand the tool should provide a possibility to run all possible checks locally, without requesting for external services.
Also it has to be pretty simple in usage, like one-two words command without lots of individual manual settings.

### PR DESCRIPTION
**Pull request by sergei-solonitcyn**
## What
Solves #8278

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,140 @@
+root = true
```
Comment (keu) [inline .editorconfig:1]: what is the purpose of this?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+[tool.black]
```
Comment (keu) [inline pyproject.toml:1]: I think we should move these file from the root folder, maybe `/python/` folder

File: tasks.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    runner = Popen("bash", shell=True, stdin=PIPE, stdout=PIPE, stderr=PIPE)
+
+    venv_name = tempfile.mkdtemp(dir=os.curdir)
+    commands = [
+        f"virtualenv {venv_name}",
+        f"source {os.path.join(venv_name, 'bin', 'activate')}",
+    ]
+    if multi_envs:
+        commands.extend([
+            "pip install -r requirements.txt",
```
Comment (keu) [inline tasks.py:96]: we are moving away from `requirements.txt`

File: .editorconfig (hunk 1 above)
Comment (sergei-solonitcyn) [reply .editorconfig:1]: It helps to apply some code standarts automatically, for developers who use one of popular IDEs in their work.

**Review by htrueman — CHANGES_REQUESTED** [review]
Agree with prev reviewers

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+[*]
+charset = utf-8
+indent_size = 4
```
Comment (tuliren) [inline .editorconfig:5]: Why is this set to `4`? We have `2` everywhere. This messes up the IDE.

Comment (sergei-solonitcyn) [reply .editorconfig:5]: Sorry, fixed it here:
https://github.com/airbytehq/airbyte/pull/9235

Comment (tuliren) [reply .editorconfig:5]: Cool. Thank you.

**Pull request merged**